core/src/main/java/com/linecorp/armeria/server/logging/structured/ServiceLogLoggingService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging.structured;
+
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.Response;
+import com.linecorp.armeria.common.logging.MessageLogConsumer;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.ResponseLog;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.logging.LogCollectingService;
+
+/**
+ * A decorating service which provides support of structured and optionally externalized request/response
+ * content logging.
+ */
+public class ServiceLogLoggingService<I extends Request, O extends Response, E>
```

Comment:
then we rename all "ServiceLog" occurrences as well? ServiceLogEntry, ApacheThriftServiceLogEntry, etc...